tsMuxer/ac3Codec.cpp

```diff
@@ -119,8 +119,8 @@ int AC3Codec::parseHeader(uint8_t* buf, uint8_t* end)
     m_bsid = id;
     if (m_bsid > 10)  // bsid = 16 => EAC3
     {
-        unsigned int numblkscod, strmtyp, substreamid, number_of_blocks_per_syncframe;
-        uint8_t acmod, lfeon, bsmod, fscod, dsurmod, pgmscle, extpgmscle, mixdef, paninfoe;
+        int numblkscod, strmtyp, substreamid, number_of_blocks_per_syncframe;
+        int acmod, lfeon, bsmod, fscod, dsurmod, pgmscle, extpgmscle, mixdef, paninfoe;
```

Comment:
These variables originate from the bitstream itself (as far as I see), thus they're most probably bitfields and should be treated as unsigned.

Reply:
No real reason to use `unsigned`:
`numblkscod` is 2 bits, `strmtyp` is 2 bits, `substreamid` is 3 bits, `number_of_blocks_per_syncframe` is <6, `acmod` is 3 bits, `lfeon` is 1 bit, `bsmod` is 3 bits, `fscod` is 2 bits, `dsurmod` is 2 bits, `pgmscle` is 1 bit, `extpgmscle` is 1 bit, `mixdef` is 2 bits, `paninfoe` is 1 bit.

But if you think `unsigned `is better than `int` in this case, I'll change.